test/color/test_raw.py

```diff
@@ -196,3 +196,87 @@ def test_module(self, device, dtype):
         raw_ops = kornia.color.RgbToRaw(kornia.color.raw.CFA.BG).to(device, dtype)
         raw_fcn = kornia.color.rgb_to_raw
         assert_close(raw_ops(img), raw_fcn(img, kornia.color.raw.CFA.BG))
+
+
+class TestRawToRgb2x2Downscaled(BaseTester):
+    def test_smoke(self, device, dtype) -> None:
+        C, H, W = 1, 4, 6
+        img = torch.rand(C, H, W, device=device, dtype=dtype)
+        assert isinstance(kornia.color.raw_to_rgb_2x2_downscaled(img, kornia.color.CFA.BG), torch.Tensor)
+
+    @pytest.mark.parametrize(
+        "batch_size, height, width",
+        [((1,), 6, 4), ((), 6, 4), ((), 0, 0), ((4, 5), 0, 0), ((2,), 2, 4), ((3,), 4, 2), ((4, 2, 1), 4, 2)],
+    )
+    def test_cardinality(self, device, dtype, batch_size, height, width):
+        img = torch.ones(*batch_size, 1, height, width, device=device, dtype=dtype)
+        assert kornia.color.raw_to_rgb_2x2_downscaled(img, kornia.color.CFA.BG).shape == (
+            *batch_size,
+            3,
+            height // 2,
+            width // 2,
+        )
+
+    def test_exception(self, device, dtype):
+        with pytest.raises(TypeError):
+            assert kornia.color.raw_to_rgb_2x2_downscaled([0.0], kornia.color.CFA.BG)
```

Comment:
don't need these assert here -- you should be asserting the error messages here, example here: https://github.com/kornia/kornia/blob/master/CONTRIBUTING.md#coding-standards

```suggestion
        with pytest.raises(TypeError) as errinf:
            kornia.color.raw_to_rgb_2x2_downscaled([0.0], kornia.color.CFA.BG)
        assert 'Input type is not a torch.Tensor. Got' in str(errinf)
```

Reply:
Done